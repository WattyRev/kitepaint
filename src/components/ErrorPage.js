import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { H1, P } from "../theme";

const StyleWrapper = styled.div`
  text-align: center;
`;

const ErrorPage = ({ errorCode, errorMessage }) => (
  <StyleWrapper>
    <H1>{errorCode}</H1>
    <P>{errorMessage}</P>
  </StyleWrapper>
);

ErrorPage.defaultProps = {
  errorCode: 404,
  errorMessage: "Page not found"
};

ErrorPage.propTypes = {
  errorCode: PropTypes.number.isRequired,
  errorMessage: PropTypes.string.isRequired
};

export default ErrorPage;
