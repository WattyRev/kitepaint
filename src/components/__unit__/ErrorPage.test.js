import React from "react";
import { shallow } from "enzyme";
import ErrorPage from "../ErrorPage";

describe("ErrorPage", () => {
  it("renders", () => {
    expect.assertions(1);
    const wrapper = shallow(<ErrorPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
