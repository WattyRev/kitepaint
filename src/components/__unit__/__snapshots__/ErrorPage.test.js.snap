// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorPage
    errorCode={404}
    errorMessage="Page not found"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          404
        </ForwardRef>,
        <P>
          Page not found
        </P>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": 404,
        },
        "ref": null,
        "rendered": 404,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  font-size: 2em;
",
            ],
          },
          "displayName": "Styled(H1)",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": [Function],
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Page not found",
        },
        "ref": null,
        "rendered": "Page not found",
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-gqjmRU",
        "isStatic": true,
        "rules": Array [
          "
  text-align: center;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-gqjmRU",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            404
          </ForwardRef>,
          <P>
            Page not found
          </P>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": 404,
          },
          "ref": null,
          "rendered": 404,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  font-size: 2em;
",
              ],
            },
            "displayName": "Styled(H1)",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Page not found",
          },
          "ref": null,
          "rendered": "Page not found",
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-gqjmRU",
          "isStatic": true,
          "rules": Array [
            "
  text-align: center;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-gqjmRU",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
